tests/KeyFrame.cpp

```diff
@@ -506,3 +513,195 @@ TEST_CASE( "Point_Vector_Constructor", "[libopenshot][keyframe]" )
+
+}
+
+
+TEST_CASE( "TrackedObjectBBox SetJson", "[libopenshot][keyframe]" )
+{
+	TrackedObjectBBox kfb;
+
+	kfb.AddBox(1, 10.0, 10.0, 100.0, 100.0, 0.0);
+	kfb.AddBox(10, 20.0, 20.0, 100.0, 100.0, 0.0);
+	kfb.AddBox(20, 30.0, 30.0, 100.0, 100.0, 0.0);	
+	kfb.AddBox(30, 40.0, 40.0, 100.0, 100.0, 0.0);
+
+	kfb.scale_x.AddPoint(1, 2.0);
+	kfb.scale_x.AddPoint(10, 3.0);
+
+	kfb.SetBaseFPS(Fraction(24.0, 1.0));
+
+	auto dataJSON = kfb.Json();
+	TrackedObjectBBox fromJSON_kfb;
+	fromJSON_kfb.SetJson(dataJSON);
+
+	CHECK(kfb.GetBaseFPS().num == fromJSON_kfb.GetBaseFPS().num);
+	
+	double time_kfb = kfb.FrameNToTime(1, 1.0);
+	double time_fromJSON_kfb = fromJSON_kfb.FrameNToTime(1, 1.0);
+	CHECK(time_kfb == time_fromJSON_kfb);
+
+	BBox kfb_bbox =  kfb.BoxVec[time_kfb];
+	BBox fromJSON_bbox = fromJSON_kfb.BoxVec[time_fromJSON_kfb];
+
+	CHECK(kfb_bbox.cx == fromJSON_bbox.cx);
+	CHECK(kfb_bbox.cy == fromJSON_bbox.cy);
+	CHECK(kfb_bbox.width == fromJSON_bbox.width);
+	CHECK(kfb_bbox.height == fromJSON_bbox.height);
+	CHECK(kfb_bbox.angle == fromJSON_bbox.angle);
+}
+
+TEST_CASE( "TrackedObjectBBox scaling", "[libopenshot][keyframe]" )
+{
+	TrackedObjectBBox kfb;
+
+	kfb.AddBox(1, 10.0, 10.0, 10.0, 10.0, 0.0);
+	kfb.scale_x.AddPoint(1.0, 2.0);
+	kfb.scale_y.AddPoint(1.0, 3.0);
+	
+	BBox bbox = kfb.GetBox(1);
+	
+	CHECK(bbox.width == 20.0);
+	CHECK(bbox.height == 30.0);
+}
+
+TEST_CASE( "AttachToObject", "[libopenshot][keyframe]" )
+{
+	std::stringstream path1, path2;
+	path1 << TEST_MEDIA_PATH << "test.avi";
+	path2 << TEST_MEDIA_PATH << "run.mp4";
+
+	// Create Timelime
+	Timeline t(1280, 720, Fraction(25,1), 44100, 2, ChannelLayout::LAYOUT_STEREO);
+
+	// Create Clip and add it to the Timeline
+	Clip clip(new FFmpegReader(path1.str()));
+	clip.Id("AAAA1234");
+
+	// Create a child clip and add it to the Timeline
+	Clip childClip(new FFmpegReader(path2.str()));
+	childClip.Id("CHILD123");
+
+	// Add clips to timeline
+	t.AddClip(&childClip);
+	t.AddClip(&clip);
+
+	// Create tracker and add it to clip
+	Tracker tracker;
+	clip.AddEffect(&tracker);
+
+	// Save a pointer to trackedData
+	std::shared_ptr<TrackedObjectBBox> trackedData = tracker.trackedData;
+
+	// Change trackedData scale
+	trackedData->scale_x.AddPoint(1, 2.0);
+	CHECK(trackedData->scale_x.GetValue(1) == 2.0);
+
+	// Tracked Data JSON
+	auto trackedDataJson = trackedData->JsonValue();
+
+	// Get and cast the trakcedObject
+	auto trackedObject_base = t.GetTrackedObject("None");
+	std::shared_ptr<TrackedObjectBBox> trackedObject = std::static_pointer_cast<TrackedObjectBBox>(trackedObject_base);
+	CHECK(trackedObject == trackedData);
+
+	// Set trackedObject Json Value
+	trackedObject->SetJsonValue(trackedDataJson);
+
+	// Attach childClip to tracked object
+	std::string tracked_id = trackedData->Id();
+	childClip.Open();
+	childClip.AttachToObject(tracked_id);
+	
+	std::shared_ptr<TrackedObjectBBox> trackedTest = std::static_pointer_cast<TrackedObjectBBox>(childClip.GetAttachedObject());
```

Comment:
Ditto, re: line length and `dynamic_pointer_cast`.
```suggestion
	auto trackedTest = std::make_shared<TrackedObjectBBox>();
	trackedTest = std::dynamic_pointer_cast<TrackedObjectBBox>(childClip.GetAttachedObject());
```